### ISSUE TITLE
Triodos Bankieren NL

### ISSUE THREAD

**Issue by ph00lt0** [PrivSec-dev/banking-apps-compat-report#133]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### App name

Triodos Bankieren NL

### Link to app

https://play.google.com/store/apps/details?id=com.triodos.bankingnl

### App version

3.39.0

### Country of the app

Netherlands

### Build Number

TP1A.220624.014.2022082400

### Device

- [ ] Pixel 6a
- [ ] Pixel 6 Pro
- [ ] Pixel 6
- [X] Pixel 5a
- [ ] Pixel 5
- [ ] Pixel 4a (5G)
- [ ] Pixel 4a
- [ ] Pixel 4 XL
- [ ] Pixel 4
- [ ] Pixel 3a XL
- [ ] Pixel 3a
- [ ] Pixel 3 XL
- [ ] Pixel 3

### Google Play installed

- [ ] Installed
- [X] Not installed

### Google Play services Network permission revoked?

- [ ] Revoked
- [ ] Not revoked
- [X] I did not have Google Play services installed

### SafetyNet Enforcement

- [ ] Enforced
- [X] Not enforced
- [ ] Unsure

### Native code debugging

- [ ] Enabled
- [X] Disabled

### Stock OS compatibility

- [ ] Works
- [ ] Does not work
- [X] Not tested

### Profile app tested in

- [X] Main user profile
- [ ] Secondary user profile
- [ ] I used a work profile device manager app (Please mention more details about it down below like the app)

### Description of the app's functionality

All functionality works as expected

**Comment by akc3n** [comment]
Thank you @ph00lt0

**Comment by spring-onion** [comment]
@readefries Try disabling secure app spawning, don't forget to reboot afterwards.

**Comment by nberlee** [comment]
I have the same problem. I tried to secure app spawning rebooted and enabled exploit protiection compatibility mode.

Still the same redirect as described by @readefries unfortunately.

**Comment by spring-onion** [comment]
Any notification about play integrity usage?

**Comment by floatingatoms** [comment]
Having the same issue. Chat customer service representative said they do not support alternative Android operating systems but that I could email info@triodos.nl about this. 

Wondering if we could collectively write a technically sound email requesting them to support GrapheneOS directly as detailed in the [detailed guide](https://grapheneos.org/articles/attestation-compatibility-guide)?

**Comment by readefries** [comment]
In Google Play, Triodos replied stating the issue is known and they're workimg on a solution

![Image](https://github.com/user-attachments/assets/90ba41fe-a016-4fc2-969a-aae58aeed82b)

**Comment by readefries** [comment]
Ah they released a new update with a vague description "bug fixes". Which doesn't seem to open the warning page 🎉  Feels like a roll-back of the broken build. Where broken is not free of any opinion 😜

**Comment by mennovogel** [comment]
Hi GrapheneOS users,
 
Let me start by saying that we're sorry that you had to deal with a version of our app that didn't work.
 
I'm happy to see that there is such an active community that helps eachother out with any issues related to GrapheneOS and the Triodos Banking app. I appreciate that you are taking the next step in controlling your security and privacy by installing GrapheneOS. And thank you so much for all the information you've sent!
 
We've heard you, and we indeed had to revert some of our changes to get it fixed again. Although we do not officially support alternative Android operating systems we do try to support them within our possibilities and within the bounds of keeping our app secure.
 
I can't go into the details of what was happening and how we solved it, but I can share that we've been looking into it and we've resolved the issue. I've also installed GrapheneOS on my test device to make sure that it doesn't happen again.
 
Oh, and yes, we do still have to update the information about rooting, because it is indeed not correct.
 
Best regards,
 
Menno Vogel
Android Developer at Triodos Bank

**Comment by readefries** [comment]
Hey Menno,

thanks for this, really appreaciated!
And just to emphasize: I understand the need for checking for rooted devices and checking for signs for insecure set-ups.

Cheers, Hendrik

**Comment by LichtZinnig** [comment]
> Hi GrapheneOS users,
> 
> Let me start by saying that we're sorry that you had to deal with a version of our app that didn't work.
> 
> I'm happy to see that there is such an active community that helps eachother out with any issues related to GrapheneOS and the Triodos Banking app. I appreciate that you are taking the next step in controlling your security and privacy by installing GrapheneOS. And thank you so much for all the information you've sent!
> 
> We've heard you, and we indeed had to revert some of our changes to get it fixed again. Although we do not officially support alternative Android operating systems we do try to support them within our possibilities and within the bounds of keeping our app secure.
> 
> I can't go into the details of what was happening and how we solved it, but I can share that we've been looking into it and we've resolved the issue. I've also installed GrapheneOS on my test device to make sure that it doesn't happen again.
> 
> Oh, and yes, we do still have to update the information about rooting, because it is indeed not correct.
> 
> Best regards,
> 
> Menno Vogel Android Developer at Triodos Bank

Hi Menno,

Just wanted to say how impressed I am with the way you handled this issue — not only was it resolved quickly, but the clear, transparent communication about it is truly exceptional. It's rare (and refreshing!) to see a bank communicate so openly and thoughtfully with its customers. Well done — this really sets Triodos apart.

Best regards,
Lucas

**Comment by thestinger** [comment]
@sand3rk /e/ doesn't keep the standard Android privacy/security model intact and doesn't keep up with basic privacy/security patches. See https://discuss.grapheneos.org/d/24134-devices-lacking-standard-privacysecurity-patches-and-protections-arent-private. It's exactly what many of these apps are trying to forbid with the Play Integrity API with GrapheneOS as collateral damage despite being far more secure than any OS approved by the Play Integrity API.

**Comment by MvRoo** [comment]
> 💜 I'm also on version 687, and I don't experience this. There is no integrity check warning. I can just use it, and I have made a few transactions with it this morning.

ah hello, colleague! interesting that you don't have the issue. could you share some more details about your setup? are you running play services? any special settings on the app?

**Comment by mennovogel** [comment]
We do our best to support GrapheneOS (and also /e/OS @sand3rk) and have at least one test device with it. We do have checks in place to check if the device is rooted. Unfortunately we sometimes see false-positives meaning that an unrooted device is detected as rooted. These false-positives can sometimes be triggered by the way an OS is installed and how the bootloader is locked/unlocked.
We're hoping to share some more information on the website in the near future.

**Comment by readefries** [comment]
> > 💜 I'm also on version 687, and I don't experience this. There is no integrity check warning. I can just use it, and I have made a few transactions with it this morning.
> 
> ah hello, colleague! interesting that you don't have the issue. could you share some more details about your setup? are you running play services? any special settings on the app?

I haven't changed any of the default app settings of GraphenOS. I do have the sandboxed Play Services installed. I must say, I enabled the optional security patches in the OS update section. You know, the one that cannot be part of the main build, due to the sources not yet being shared by Google, as a result of the new update cycle, which gives other vendors more time to patch them.

**Comment by Raubion** [comment]
Dear @mennovogel 

Thanks so much for making it possible to use the Triodos app on GrapheneOS. I made the switch yesterday and it works like a charm.

I used to use CalyxOS, but this summer started to get the "your device is rooted" popups. I first rolled back to version 570 and then switched to IodeOS when this version stopped working. At first I wanted to keep using an Android version with a microg implementation. When I also started to get the notification on IodeOS, I decided I needed to look further. The switch to GrapheneOS has now led me to a functional Triodos app. With the other OSes, something somehow goes wrong with Builds including the Play Integrity Patch (see for more [here](https://lineage.microg.org/)).

I do, however, wish to underscore the need for a better error message. Indeed, my phone was never rooted and had a locked bootloader. But more importantly, I want to point to the importance of letting customers that get in touch with the customer service of Triodos know about this thread or the article on [security.nl](https://www.security.nl/posting/897273/Triodos+laat+bank-app+na+aanpassingen+weer+op+GrapheneOS+werken) which I found yesterday (and which led me here). Somehow this also didn't come up previously when searching. 

I've gotten in touch three times with customer service since the summer and each time somehow been told that alternative Android versions or stores are not supported. And as much as I understand this position from a security perspective, being pointed to this thread would have saved me se hassle. 

I'm very happy with the support, with the communication in this thread and with the fact that the app now works again. Just a pointer to help some fellow customers struggling with a disfunctional app.

Yours

**Comment by ce-designs** [comment]
> I do see a warning now stating Play Services requires phone access, would be nice if this was optional IMHO

Same here. I found the following information about the phone access of play services:

_**To improve security**
Google Play services may periodically collect your phone information in the background to improve the security and reliability of digital wallet and payment services._

Not sure how collecting my phone information contributes to improved security. 
Anyone care to explain this? @mennovogel ?

**Comment by readefries** [comment]
I 'solved' it by hiding this specific notification in notification settings :D

**Comment by ce-designs** [comment]
> I 'solved' it by hiding this specific notification in notification settings :D

Yeah you 'solved' it 😄, but with hiding the notification you still don't know how this affects the app. Perhaps it will work fine without play services having access to phone, but I prefer to know the consequences and I'd still like to know how collecting phone information contributes to improved security?

**Comment by Geblaat** [comment]
@mennovogel 
Can the app support Memory Tagging? It has worked fine for a short while with Memory Tagging enabled, but the last few months it started crashing again with Memory Tagging enabled.

**Comment by readefries** [comment]
> > I 'solved' it by hiding this specific notification in notification settings :D
> 
> Yeah you 'solved' it 😄, but with hiding the notification you still don't know how this affects the app. Perhaps it will work fine without play services having access to phone, but I prefer to know the consequences and I'd still like to know how collecting phone information contributes to improved security?

The app doesn't really need these permissions, as it can simply use the [call intent](https://developer.android.com/reference/app-actions/built-in-intents/communications/create-call) to accomplish this.

**Comment by Confined164** [comment]
I can't get the app (version 4.55.0) to work on Graphene OS. After a couple of seconds I am redirected to the rooted FAQ site. Customer service tells me only Apple and stock android are supported.

**Comment by readefries** [comment]
4.55.0 (also 4.56.1) is working fine on Graphene OS as long as you use the signed versions built by Graphene. The fingerprint of your own build can't be verified as the hashes of it our unknown.

**Comment by MvRoo** [comment]
> 4.55.0 (also 4.56.1) is working fine on Graphene OS as long as you use the signed versions built by Graphene. The fingerprint of your own build can't be verified as the hashes of it our unknown.

Indeed. I actually confirmed that today. The Triodos app redirects to the 'you are rooted' site on a GrapheneOS ROM signed with my own signing key, while it doesn't on a ROM signed by Graphene.

**Comment by jordischoots** [comment]
I'm unable to get any version working on my Pixel 10 Pro. The same setup works fine on my Pixel 7, so it seems device-specific. On the 10 pro I get redirected to the rooted FAQ. Any idea what might cause this?

**Comment by MvRoo** [comment]
> I'm unable to get any version working on my Pixel 10 Pro. The same setup works fine on my Pixel 7, so it seems device-specific. On the 10 pro I get redirected to the rooted FAQ. Any idea what might cause this?

Are you running the same version of grapheneos on both, in terms of build number and signing key?

**Comment by jordischoots** [comment]
> 
> Are you running the same version of grapheneos on both, in terms of build number and signing key?

Build numer is different, both are signed by Grapheneos.

**Comment by spring-onion** [comment]
@mennovogel It sounds like you haven't added our keys for the Pixel 10 series yet.

**Comment by nicolasmaia** [comment]
Does Triodos's web banking require the Android app for 2FA?

**Comment by ce-designs** [comment]
> Does Triodos's web banking require the Android app for 2FA?

No, registration of your android/iOS device can only be done by using the official Triodos identifier device or by uploading an photo (using the app) of your passport or identification card. No options 2FA login available once your device is registered. At least not for as far as I know.
